### PR TITLE
[event_engine][fork] Don't delete object

### PR DESCRIPTION
**Pull request by eugeneo**
Fixes use-after-free

### CODE REVIEW

**Comment by drfloob** [comment]
For posterity: need to establish that these won't be leaked under all scenarios.

File: src/core/client_channel/subchannel_pool_interface.cc [hunk 1]
```diff
@@ -55,8 +55,7 @@ std::string SubchannelKey::ToString() const {
   auto addr_uri = grpc_sockaddr_to_uri(&address_);
   return absl::StrCat(
       "{address=",
-      addr_uri.ok() ? addr_uri.value() : addr_uri.status().ToString(),
-      ", args=", args_.ToString(), "}");
+      addr_uri.ok() ? addr_uri.value() : addr_uri.status().ToString(), "}");
```
Comment (markdroth) [inline src/core/client_channel/subchannel_pool_interface.cc:58]: The subchannel args are part of the key, so they're important to log here.  Please revert this change.

File: src/core/client_channel/subchannel.cc [hunk 2]
```diff
@@ -697,8 +698,9 @@ void Subchannel::OnRetryTimer() {
 
 void Subchannel::OnRetryTimerLocked() {
   if (shutdown_) return;
-  gpr_log(GPR_INFO, "subchannel %p %s: backoff delay elapsed, reporting IDLE",
-          this, key_.ToString().c_str());
+  gpr_log(GPR_INFO,
```
Comment (markdroth) [inline src/core/client_channel/subchannel.cc:701]: This does not seem like a good change.  The subchannel key is useful to log here, and the pid and tid are not.  Please revert this change.

File: src/core/client_channel/subchannel.cc [hunk 3]
```diff
@@ -743,10 +745,10 @@ void Subchannel::OnConnectingFinishedLocked(grpc_error_handle error) {
     const Duration time_until_next_attempt =
         next_attempt_time_ - Timestamp::Now();
     gpr_log(GPR_INFO,
-            "subchannel %p %s: connect failed (%s), backing off for %" PRId64
-            " ms",
-            this, key_.ToString().c_str(), StatusToString(error).c_str(),
-            time_until_next_attempt.millis());
+            "pid %d subchannel %p %s: connect failed (%s), backing off "
+            "for %" PRId64 " ms",
+            getpid(), this, key_.ToString().c_str(),
```
Comment (markdroth) [inline src/core/client_channel/subchannel.cc:750]: There is no need to add the pid here.  Please revert.

File: src/core/client_channel/subchannel.cc [hunk 4]
```diff
@@ -765,6 +767,9 @@ void Subchannel::OnConnectingFinishedLocked(grpc_error_handle error) {
             self.reset();
           }
         });
+  } else {
+    gpr_log(GPR_INFO, "Connected subchannel %p %s, pid %d", this,
```
Comment (markdroth) [inline src/core/client_channel/subchannel.cc:771]: This is going to be far too noisy.  Please remove.

File: src/core/handshaker/handshaker.cc [hunk 5]
```diff
@@ -136,6 +138,8 @@ bool HandshakeManager::CallNextHandshakerLocked(grpc_error_handle error) {
     }
     // Cancel deadline timer, since we're invoking the on_handshake_done
     // callback now.
+    gpr_log(GPR_INFO, "Canceling deadline id: %p, pid: %d",
```
Comment (markdroth) [inline src/core/handshaker/handshaker.cc:141]: Please revert all of the changes in this file.

**Review by markdroth — COMMENTED** [review]
I don't think we want any of the changes in the subchannel or handshaker code.  Please revert them.

**Pull request closed without merging**